src/solvers/dgsem_p4est/containers_2d_manifold_in_3d_covariant.jl

```diff
@@ -0,0 +1,209 @@
+                                                                                         2)
+@inline Base.ndims(::P4estInterfaceContainerVariableCoefficient{NDIMS}) where {NDIMS} = NDIMS
+
+# Create interface container and initialize interface data.
+function Trixi.init_interfaces(mesh::Union{P4estMesh, T8codeMesh},
+                               equations::AbstractCovariantEquations, basis, elements)
+    NDIMS = ndims(elements)
+    uEltype = eltype(elements)
+
+    # Initialize container
+    n_interfaces = Trixi.count_required_surfaces(mesh).interfaces
+
+    _u = Vector{uEltype}(undef,
+                         2 * nvariables(equations) * nnodes(basis)^(NDIMS - 1) *
+                         n_interfaces)
+    u = Trixi.unsafe_wrap(Array, pointer(_u),
+                          (2, nvariables(equations),
+                           ntuple(_ -> nnodes(basis), NDIMS - 1)...,
+                           n_interfaces))
+
+    _auxiliary_variables = Vector{uEltype}(undef,
+                                           2 * nauxvars(equations) *
+                                           nnodes(basis)^(NDIMS - 1) *
+                                           n_interfaces)
+    auxiliary_variables = Trixi.unsafe_wrap(Array, pointer(_auxiliary_variables),
+                                            (2, nauxvars(equations),
+                                             ntuple(_ -> nnodes(basis), NDIMS - 1)...,
+                                             n_interfaces))
+
+    _neighbor_ids = Vector{Int}(undef, 2 * n_interfaces)
+    neighbor_ids = Trixi.unsafe_wrap(Array, pointer(_neighbor_ids), (2, n_interfaces))
+
+    _node_indices = Vector{NTuple{NDIMS, Symbol}}(undef, 2 * n_interfaces)
+    node_indices = Trixi.unsafe_wrap(Array, pointer(_node_indices), (2, n_interfaces))
+
+    interfaces = P4estInterfaceContainerVariableCoefficient{NDIMS, uEltype, NDIMS + 2}(u,
+                                                                                       auxiliary_variables,
+                                                                                       neighbor_ids,
+                                                                                       node_indices,
+                                                                                       _u,
+                                                                                       _auxiliary_variables,
+                                                                                       _neighbor_ids,
+                                                                                       _node_indices)
+
+    Trixi.init_interfaces!(interfaces, mesh)
+
+    return interfaces
+end
+
+# Initialize node_indices of interface container (this is copied from Trixi.jl)
+@inline function Trixi.init_interface_node_indices!(interfaces::P4estInterfaceContainerVariableCoefficient{2},
```

Comment:
I would also prefer to avoid duplicated code, but in Julia, you cannot make a subtype of a non-abstract type. I think the solution to this (and overall a more elegant approach) is to put all the auxvars stuff into its own cache, separate from the existing element and interface containers (I believe this was suggested by @ranocha).

Reply:
Putting the auxiliary stuff into its own cache avoided this duplication.